From 385da566017e62ff26dd83905dc9ced3e0b19bff Mon Sep 17 00:00:00 2001
From: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
Date: Mon, 12 Dec 2022 14:48:29 +0700
Subject: [PATCH 13/16] include: configs: rcar-gen3-common.h: W/A: Second
 starting USB if not available

Although using 'usb_pgood_delay" can prevent USB can not be detected
issue. We may get this issue again when restarting from Linux.

Use this W/A to prevent this issue.

Signed-off-by: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
---
 include/configs/rcar-gen3-common.h | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/include/configs/rcar-gen3-common.h b/include/configs/rcar-gen3-common.h
index b4cdf1f1fd..e8369b47a4 100644
--- a/include/configs/rcar-gen3-common.h
+++ b/include/configs/rcar-gen3-common.h
@@ -66,6 +66,9 @@
 	"fdtfile=r8a774a1-hihope-rzg2m.dtb\0" \
 	"kernel_addr_r=0x48080000\0" \
 	"boot_efi_binary=efi/boot/"BOOTEFI_NAME"\0" \
+	"scan_for_usb_dev=" \
+		"usb start; " \
+		"if test ! -e usb ${devnum}:1 /; then usb reset; fi;\0" \
 	"scan_boot_efi=" \
 		"part list ${devtype} ${devnum} devplist; "  \
 		"env exists devplist || setenv devplist 1; " \
@@ -86,14 +89,14 @@
 			"setenv devtype mmc;" \
 			"run scan_boot_efi;\0" \
 	"usb0=" \
-			"usb start;" \
 			"setenv devnum 0;" \
 			"setenv devtype usb;" \
+			"run scan_for_usb_dev;"\
 			"run scan_boot_efi;\0"\
 	"usb1=" \
-			"usb start;" \
 			"setenv devnum 1;" \
 			"setenv devtype usb;" \
+			"run scan_for_usb_dev;"\
 			"run scan_boot_efi;\0" \
 	"boot_targets=" \
 			"mmc0 mmc1 usb0 usb1\0" \
-- 
2.17.1

